Index: lilo/lilo.new
===================================================================
--- lilo/lilo.new	(Revision 331)
+++ lilo/lilo.new	(Revision 332)
@@ -435,17 +435,20 @@
     # (i.e. disk with the PReP partition), else set force_fat=1 also
     # check for a valid and readable partition containing the config
     # file.
-    echo -n > /etc/yaboot.conf
-    local yaboot_path=$(get_of_path /etc/yaboot.conf)
-    local yaboot_part=${yaboot_path%,/*yaboot.conf}
+    echo -n >> /etc/yaboot.conf
+    dummy=$(get_of_path /etc/yaboot.conf)
+    local yaboot_part=${dummy%,/*yaboot.conf}
 
     if [ "${boot_disk}" != "${yaboot_part%%:+([0-9])}" ]; then
 	force_fat=1
     fi
-    if ! is_on_primary "$yaboot_path" ||
+    if ! is_on_primary "$yaboot_part" ||
 	[[ ${fsys[ $(device_of_file /etc/yaboot.conf)]} != @(ext2|ext3|msdos|vfat|reiserfs|iso9660|xfs) ]]; then
 	force_fat=1
     fi
+    if (( force_fat == 1 )); then
+	yaboot_part=
+    fi
 
     # starting the work and write the yaboot.conf
     {
