--- lilo-0.1.2/lilo.new.LTC16275_SUSE87050	2005-06-20 11:36:31.662209997 +0200
+++ lilo-0.1.2/lilo.new	2005-06-20 12:01:24.113469098 +0200
@@ -573,6 +573,10 @@ function running_on_chrp () {
 	# copy all kernels and initrds from non supported file systems
 	# to one FAT partition
 
+ 	MKDOSFS="-s 1 -F 16"   # default values, minimum size of 2076kB
+ 	MINDOSFSSIZE=$[2*2076] # does prevent possible misdetection of
+ 	                       #  file system as FAT12
+
 	# check size of partition OPTION_BOOT, must fit at least the default kernel
 	local -i bootsize=$(( (p_end[partition]-p_start[partition])*2 ))	# size in blocks
 	
@@ -581,6 +585,10 @@ function running_on_chrp () {
 	(( defaultsize += (vmlinux_default==0) ? 0 : $(fsize ${vmlinux_copy[$vmlinux_default]}) ));
 	(( defaultsize += (initrd_default==0) ? 0 : $(fsize ${initrd_copy[$initrd_default]}) ));
 
+	if (( defaultsize < MINDOSFSSIZE )); then
+	    defaultsize=$MINDOSFSSIZE;
+	fi
+
 	if (( bootsize < defaultsize )); then
 	    # may be resize of the partition is possible and helps?
 	    if (( (p_start[partition+1]-p_start[partition])*2  < defaultsize )); then
@@ -604,7 +612,7 @@ function running_on_chrp () {
 		error "Partition conversion failed for $OPTION_BOOT" 11
 	    fi
 	fi
-	mkfs.msdos -s 8 $OPTION_BOOT || exit 2
+	mkfs.msdos $MKDOSFS $OPTION_BOOT || exit 2
 	cd $TEMP
 	mkdir -p boot || exit 2
 	mount $OPTION_BOOT boot || exit 2
