 second/yaboot.c |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletion(-)

Index: yaboot-1.3.11/second/yaboot.c
===================================================================
--- yaboot-1.3.11.orig/second/yaboot.c
+++ yaboot-1.3.11/second/yaboot.c
@@ -796,6 +796,7 @@ yaboot_text_ui(void)
      struct boot_file_t	file;
      int			result;
      static struct boot_param_t	params;
+     void 		*claim_base;
      void		*initrd_base;
      unsigned long	initrd_size;
      void                *sysmap_base;
@@ -951,7 +952,13 @@ yaboot_text_ui(void)
 	       }
 	       else {
 #define INITRD_CHUNKSIZE 0x400000
-		    initrd_base = prom_claim(loadinfo.base+loadinfo.memsize, INITRD_CHUNKSIZE, 0);
+		    claim_base = loadinfo.base + loadinfo.memsize;
+		    for (result = 0; result < 42; result++) {
+			    initrd_base = prom_claim(claim_base, INITRD_CHUNKSIZE, 0);
+			    if (initrd_base != (void *)-1)
+				    break;
+			    claim_base += (unsigned long)claim_base + (1*1024*1024);
+		    }
 		    if (initrd_base == (void *)-1) {
 			 prom_printf("Claim failed for initrd memory\n");
 			 initrd_base = 0;
